3_summarize/src/annual_thermal_metrics.R

```diff
@@ -322,26 +348,30 @@ calc_monthly_summary_stat <- function(date, wtr_at_depth, depth_prefix, stat_typ
 }
 
 #' days in which there is any part of water column in a temperature range
-calc_days_height_vol_within_range <- function(date, depth, wtr, hypso, temp_low, temp_high) {
+calc_days_height_vol_within_range <- function(daily_data, hypso, temp_low, temp_high) {
   stopifnot(length(temp_low) == length(temp_high))
   
-  grpd_data <- tibble(date, depth, wtr) %>% 
-    group_by(date)
+  # Create a vector of names in the order we want them (height_[temp range 1], vol_[temp range 1], days_[temp range 1], etc)
+  temp_range_str <- sprintf("%s_%s", temp_low, temp_high)
+  out_colnames <- apply(expand.grid(c("height", "vol", "days"), temp_range_str), 1, paste, collapse = "_")
+  
+  setDT(daily_data)
+  
+  dt <- daily_data[, find_Z1_Z2(wtr, depth, temp_high, temp_low), by = list(year,date)][, `:=` (
```

Comment:
I'm not able to run this to test it so I'm not sure that it will fail, but you've removed `wtr` and `depth` from the function inputs but now you are trying to access them here, so I'd expect an error or (worse 😱 ) accidentally snagging those values from other spots above the scope of this function (unlikely, but possible if `wtr <- 4` or something appears somewhere above this function and not scoped in another function. 

Oh...now I see, perhaps this is data.table syntax?? ahhhh. Ignore my comment if so!

Reply:
Finally going to merge this and just want to comment to say that this is definitely data.table syntax! `wtr` and `depth` here are referring to columns within `daily_data`, so all good